Add necessary fields for beta46 experiment 2

Uberi opened the pull request:
The second beta46 experiment changed the format; there's now a system addon that sets environment/settings/e10sCohort.

This means we need a new flag in E10SExperiment that can handle this correctly, or a new class for it. This PR is an initial implementation, it probably still needs some more tests.

vitillo commented on src/main/scala/streams/E10sExperiment.scala:
```diff
@@ -24,7 +24,7 @@ private class SampleIdPartitioner extends Partitioner{
 }
 
-case class E10sExperiment(experimentId: String, prefix: String) extends DerivedStream {
+case class E10sExperiment(experimentId: String, prefix: String, newE10SExperimentFormat: Boolean) extends DerivedStream {
```
Let's just support the new experiment format and remove the past ones from `DerivedStream`.

vitillo commented on src/main/scala/streams/E10sExperiment.scala:
```diff
@@ -49,27 +49,38 @@ case class E10sExperiment(experimentId: String, prefix: String) extends DerivedS
+          val cohort = settings \ "e10sCohort"
+
+          (clientId, sampleId, cohort) match {
+            case (Some(client: String), Some(sample: Double), JString(cohort)) if cohort != "unknown" =>
```
When is cohort == unknown?

Uberi replied:
The unknown cohort shouldn't occur usually, but it means something basically went wrong. Now that I think of it, this would still be useful information to have, so I'll remove that check.

vitillo replied:
OK

vitillo commented on src/main/scala/streams/E10sExperiment.scala:
```diff
@@ -49,27 +49,38 @@ case class E10sExperiment(experimentId: String, prefix: String) extends DerivedS
+          val branch = addons \ "activeExperiment" \ "branch"
+
+          // Ignore the first session of the experiment
+          val log = parse(fields.getOrElse("payload.log", "[]").asInstanceOf[String])
```
The reason we were doing this check is that in the first session of the experiment an user might be flagged as having e10s on but effectively still have it disabled as it requires a restart. How is this handled with the new format?

Uberi replied:
According to :felipe:

    16:58 <azhang> for experiments, we had a thing where on the first session, the user is in the test group, but since they didn't restart yet, e10s isn't actually enabled for that session
    16:59 <felipe> azhang: there's a possibility, but it's much smaller with the system add-on. We looked at the data and that happened to about 0.5% of users. If that's really important to you you can use the "e10sEnabled" flag to be 100% sure
    17:03 <felipe> azhang: same reason, the code to enable them ran too late in the session to get it enabled in that session.  Since the system add-on startup is a lot earlier than the previous experiment system, this has a smaller chance of happening
    17:03 <felipe> azhang: but later when we start whitelisting some add-on users for e10s that chance might increase again. So I suggest that you take e10sEnabled into account if that's simple enough

So I'll add a check for that in a bit.

vitillo replied:
OK

vitillo commented on src/main/scala/DerivedStream.scala:
```diff
@@ -188,27 +188,32 @@ object DerivedStream {
 
         case "e10s-enabled-aurora-43" => // https://bugzilla.mozilla.org/show_bug.cgi?id=1193089
           val from = options.getOrElse('fromDate, "20151022")
-          val exp = E10sExperiment("e10s-enabled-aurora-20151020@experiments.mozilla.org", "telemetry/4/saved_session/Firefox/aurora/43.0a2/")
```
Please remove the older experiments.

Uberi commented:
So GitHub still seems to be a bit wonky, but I removed the old experiments and added a check for first session weirdness. Seems like Travis isn't updating either, so push hooks are probably down as well.

vitillo commented on src/main/scala/streams/E10sExperiment.scala:
```diff
@@ -49,25 +58,19 @@ case class E10sExperiment(experimentId: String, prefix: String) extends DerivedS
-          case (Some(client: String), Some(sample: Double), JString(id), JString(branch)) if id == experimentId && status.isEmpty =>
+        val settings = parse(fields.getOrElse("environment.settings", "{}").asInstanceOf[String])
+
+        // the e10s cohort determines which group a user belongs to in the e10s experiment (as of beta46 experiment 2; previously, we used the Telemetry Experiments framework)
```
Please limit comments to 80 characters.

vitillo commented on src/main/scala/streams/E10sExperiment.scala:
```diff
@@ -49,25 +58,19 @@ case class E10sExperiment(experimentId: String, prefix: String) extends DerivedS
+        // and sets the e10sCohort to "test", yet it is too late in the session to actually enable e10s
+        // the suggested solution is to check the value of environment/settings/e10sEnabled, and exclude those pings from consideration
+        val JBool(e10sEnabled) = settings \ "e10sEnabled"
+        val pingShouldBeExcluded = cohort == "test" && !e10sEnabled // the ping is in the "test" cohort, which means e10s should be enabled, yet e10s wasn't actually enabled
```
Could you do the same check for the control group? Might come in handy for the future.

vitillo replied:
`pingShouldBeExcluded` is never used in the code

Uberi commented:
Changes:

* Add control group check as well.
* Fix comments.

mreid-moz commented on src/main/resources/application.conf:
```diff
@@ -1,3 +1,3 @@
 app {
-    parquetBucket="telemetry-parquet"
+    parquetBucket="telemetry-test-bucket"
```
Eep, please don't change this in "master"

vitillo commented:
Could you please squash all your commits into a single one that has a clear short title and a description of your changes?

vitillo commented:
Looks good, thanks!
